YunoHost/yunohost
# Configure automatically the DNS records using lexicon

## The problem
DNS configuration is tedious to apply by hand

([oriignal issue](https://github.com/YunoHost/issues/issues/185))

## Solution
Lexicon is a library that permits to programmaticaly set DNS records with an API key provided by the registrar.

## PR Status
Work in Progress!
For now, we didn't begin to use lexicon:
Our PR permits to change some settings of the DNS generated records.

## How to test

First install lexicon on yunohost : `apt install lexicon`.

Copy `yunohost/data/other/providers_list.yml` to `/usr/share/yunohost/other/providers_list.yml`

Some commands to try:
- add a provider: `yunohost domain registrar set <domain> <registrar-name>` and answer the interactive questions
- see the generated config `yunohost domain dns-conf <domain>`
- fiddle with the settings (interface similar with `yunohost app setting`) `yunohost domain setting ...`
- once you've changed some settings or added a provider, you can read `/etc/yunohost/domains.yml` where all the settings are stored.
- see the (new) generated config `yunohost domain dns-conf <domain>`
- push your DNS configuration to your registrar (which must be previously added) `yunohost domain push_config`. Firstly, check the generated config to confirm you it's what you want, and backup your DNS entries, as this function has not been deeply tested.

Look for differences in the `actionsmap.yml` and `domains.py` to see all the new available commands! :)
